src/main/java/com/adyen/service/TerminalCloudAPI.java

```diff
@@ -83,4 +82,18 @@ public TerminalAPIResponse sync(TerminalAPIRequest terminalAPIRequest) throws IO
         }.getType());
         return terminalAPIResponse;
     }
+
+    /**
+     * Checks if SaleToAcquirerData is set. If it is not creates an encoded base 64 string with application info.
+     *
+     * @param terminalAPIRequest TerminalAPIRequest
+     * @return TerminalAPIRequest
+     */
+    private static TerminalAPIRequest setSaleToAcquirerDataModel(TerminalAPIRequest terminalAPIRequest) {
+        if (terminalAPIRequest != null && terminalAPIRequest.isSaleToAcquirerDataModelSet()) {
+            SaleToAcquirerDataModel saleToAcquirerDataModel = new SaleToAcquirerDataModel();
```

Comment:
This object will be created with ApplicationInfo set by default, but I'm a bit confused on how the other fields of SaleToAcquirerDataModel will be set (metadata, keyValueData, etc). Do we need to cover that as well?

Reply:
The SaleToAcquirerDataModel with all the other nested fields should be set by the consumer application. If for any reason it is not set then it is created by default with only the application info prefilled.

Reply:
Oh I see. And we're assuming it will already be set as a base64 string, right?

Reply:
Yes as it is mentioned [here](https://docs.adyen.com/point-of-sale/pass-application-information#base64-the-application-information)

Reply:
Cool, thanks for the explanation :)